### ISSUE TITLE
Deepfool not working on cleverhans 2.1.0

### ISSUE THREAD

**Issue by B-Yassine** [tensorflow/cleverhans#913]
I've just realized that there is a bug in the implementation of the deepfool attack on cleverhans.
I tried using the original source code implemented in pytorch and the cleverhans code with the same parameters, same environment and on the same image and the output was completely different, I contacted the author of the paper and he told that he doesn't know about the cleverhans implementation and adviced me to use his own implementation

**Comment by cihangxie** [comment]
Thanks for bring this issue. 

For the network used in pytorch code, and the network used in tensorflow code, do they have exactly the same parameter, i.e., their network structures and weights are exactly the same? If they are the same, i think there should be a bug here. If they are not exactly the same, the outputs should be completely different.

**Comment by B-Yassine** [comment]
Thanks for your reply, when you say network are you talking about the
classification's model used, for instance inception v3 or resnet etc. If so
am quite sure that it's the same as am working on the inceptionv3 for both

**Comment by cihangxie** [comment]
Yes, inceptionv3 is the network structure. Do these two inceptionv3 have the same weights? For example, by feeding the same image to these two inceptionv3, do they output the same 1000/1001-dimension logit output?

**Comment by B-Yassine** [comment]
Yep it's the same model on both

**Comment by cihangxie** [comment]
could you please provide the checkpoints of your inceptionv3 in tensorflow and pytorch?

**Comment by B-Yassine** [comment]
You can check the 2 implementations here : https://github.com/keras-team/keras-applications/blob/master/keras_applications/inception_v3.py   and  here :  https://github.com/pytorch/vision/blob/master/torchvision/models/inception.py   

The architecture is the same, taken from : Rethinking the Inception Architecture for Computer Vision

However the trainining process is not the same: keras model is using:
'inception_v3_weights_tf_dim_ordering_tf_kernels.h5'
and the Torchvision: 
model_zoo.load_url(model_urls['inception_v3_google'])

But i think it's not the case of the probleme because when using the keras model on the Cleverhans's deepfool implementation, although i augment the perturbations it seems it's not perturbing the image and nothing happens.

Here is my code:


import numpy as np
import keras
from keras import backend
import tensorflow as tf
from cleverhans.utils_keras import KerasModelWrapper
from keras.applications import inception_v3
import cleverhans.attacks
import scipy.misc

keras_model = inception_v3.InceptionV3(weights='imagenet')

# Set the learning phase to false, the model is pre-trained.
backend.set_learning_phase(False)

# Set TF random seed to improve reproducibility
tf.set_random_seed(1234)

if keras.backend.image_dim_ordering() != 'tf':
    keras.backend.set_image_dim_ordering('tf')
    print("INFO: '~/.keras/keras.json' sets 'image_dim_ordering' to "
          "'th', temporarily setting to 'tf'")

# Retrieve the tensorflow session
sess = backend.get_session()

image = scipy.misc.imread("input_image.png")

# Resizing the image to be of size 299 * 299
image = np.array(scipy.misc.imresize(image, (299, 299)),
                 dtype=np.float32)

# converting each pixel to the range [0,1] (Normalization)

image = np.array([image / 255.0])

wrap = KerasModelWrapper(keras_model)

#cw = cleverhans.attacks.CarliniWagnerL2(wrap, sess = sess)
deepfool = cleverhans.attacks.DeepFool(wrap, sess = sess)
deepfool_params = {'nb_candidate':50,
                    'overshoot':500,
                    'max_iter':500,
                    'nb_classes':None,
                    'clip_min':0.0,
                    'clip_max':1.0}

adv_deepfool = deepfool.generate_np(image, **deepfool_params)
scipy.misc.imsave("adversarial_deepfool.png", adv_deepfool[0])

**Comment by B-Yassine** [comment]
Yes, i know what you are talking about, i know that the attack will not be the same on the two models since they have maybe different weights as they were trained independently, the issue was that the deepfool attack is not working on cleverhans 2.1.0, that is why i posted the code, because i tried a plethora of deepfool's parameters differenciations and i figured out that it is really not working on the keras pretrained inceptionv3. That's my problem. Then i taught that the problem was in my code so i tried the C&W attack using roughly the same implementation and it works normally, Thanks for your time and your help

**Comment by cihangxie** [comment]
Oh, now I get your point. I will take a look. Thanks for clarifying.

**Comment by npapernot** [comment]
Sounds like this was resolved, if that's not the case, feel free to reopen or make a PR that fixes the issue.
